logic.typ

```diff
@@ -159,6 +159,50 @@
   }
 }
 
+#let fill-remaining-height(
+  margin: 0%,
+  before-label: "before-fill",
+  ..box-kwargs,
+  h-align: left,
+  content,
+) = {
+  locate(loc => {
+    let prev = query(selector(label("before-fill")).before(loc), loc)
+    if prev.len() == 0 {
+      panic("You must use the label `" + before-label + "` before calling this function")
+    }
+    let prev-pos = prev.last().location().position()
+    let loc-page = loc.position().page
+    if prev-pos.page != loc-page {
+      panic(
+        "Label `" + before-label + "` must appear on the same page as the content passed"
+        + " to this function (page " + str(loc-page) + "), but the last found label"
+        + " was on page " + str(prev-pos.page)
+      )
+    }
+    layout(page-size => {
+      let kwargs = box-kwargs.named()
+      let initial-width = kwargs.at("width", default: page-size.width)
+      if type(initial-width) == "ratio" {
+        initial-width = initial-width * page-size.width
+      }
+      kwargs.insert("width", initial-width)
+      // let width = calc.max(page-size.width, kwargs.at("width", default: 0pt))
+      let remaining-height = (page-size.height - prev-pos.y) * (100% - margin)
```

Comment:
Shouldn't it be a `-` instead of a `*` here?

Reply:
```
error: cannot divide these two lengths
    ┌─ \logic.typ:200:22
    │
200 │         let h-ratio = remaining-height / boxed-size.height
    │                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

This is what I got when passing a sized unit (pt, em, etc.) instead of a ratio, so I decided to treat `margin` as a percentage. I can add an assert for this to clarify any confusing messages to users

Reply:
The issue is that you canmot simply divide fixed lengths (pt, cm, in,...) by font size-depending lengths (em). So margins in pt, cm etc. should work. If we move the calculation of the remaining height into `style`, we can also add
```typ
let margin = measure(h(margin), style).width
```
to get the margin in pt, regardless of the input.

And then `remaining-height` should be
```typ
container-size - prev-pos.y - 2 * margin
```
or am I getting something wrong? (`2 *` because of top and bottom margin).

Reply:
Good point, thanks! Usually typst has great error messages, but I wasn't sure for the reason behind this one. I fixed the logic, but left out the factor of 2 since the content is top-aligned by default. I could rename the parameter to "bottom-margin" for clarity if desired. I would only want it horizontally centered if the content stretched to the full extent of the box, similar to ppt's behavior. That is harder to replicate using `scale` alone